src/System Application/App/Rest Client/src/HttpContentImpl.Codeunit.al

```diff
@@ -32,9 +116,9 @@ codeunit 2355 "Http Content Impl."
     var
         Headers: HttpHeaders;
     begin
-        if not HttpContent.GetHeaders(Headers) then begin
-            HttpContent.Clear();
-            HttpContent.GetHeaders(Headers);
+        if not this.HttpContent.GetHeaders(Headers) then begin
```

Comment:
> echoing a comment made on Viva Engage: [Peter D. Jørgensen](https://www.yammer.com/dynamicsnavdev/#/users/1869009403904) – [36 minutes ago](https://www.yammer.com/dynamicsnavdev/#/Threads/show?threadId=3085674977689600)
> 
> Why are the global variables addressed as "this." ?
> 
> Should we also start doing that, when calling functions in the current object?

It's a good point, and I hoped to force a discussion by using it. Seems to work. 😉

The first reason to use the keyword `this` for referencing globals in the same object is that it's mentioned in the documentation: https://learn.microsoft.com/en-us/dynamics365/release-plan/2024wave2/smb/dynamics365-business-central/use-this-keyword-codeunit-self-reference#feature-details. There is even a Codecop rule AA0248 with an AL code action. The documentation also states that the System Application has been modified to use `this`.

The second reason is that I started by using the same name for parameters and global variables and then distinguished by using `this` to refer to the global variable. This is an example where `this` really would make sense. But that failed on Codecop rule AA0244, and I had no choice but to rename the parameters. But I kept `this` for referencing global variables.

I'm still not sure if this is really what we want or what problem it actually solves. It would help when parameter names are the same as global variables, but the Codecop rule already prevented that.

Reply:
Yeah, I'm not getting in the middle of this discussion 🤣 My take has always been: if it improves code readability, use it. Use your common sense. If your code get unnecessarily cluttered, then don't use it. Unfortunately that doesn't always work with _rules_, which so many are so fond of. I hence tend to get into trouble with my view on how to use such things 😝 So I'll leave it to others to decide on the rules, and then I'll blindly follow them!

Reply:
If you are getting hit by AA0244, you must rename the variable. The code is not readable. 
We do not use this in the code right now. Usage of this is an exception. We should either apply this codeword to all codebase or use it very sparsely.